test/emmy/expression/compile_test.cljc

```diff
@@ -157,15 +153,15 @@
             (with-redefs [gensym (fn
                                    ([] (clojure.core/gensym))
                                    ([x] x))]
-              (is (= #?(:clj `(fn [~'x] (+ (* -1.0 ~'x) 28.0))
+              (is (= #?(:clj `(fn [~'x] (+ (* -1 ~'x) 28.0))
```

Comment:
I am still worried here... see this:

```clj
(set! *unchecked-math* :warn-on-boxed)

;; The first two emit warnings, the next two do not.
(let [f1 (fn [x] (+ (* -1 x) 28.0))
      f2 (fn [x] (+ (* -1.0 x) 28.0))
      f3 (fn [^double x] (+ (* -1.0 x) 28.0))
      f4 (fn [^double x] (+ (* (double -1) x) 28.0))
      n 10000000000]  
  (time (dotimes [_ n]
          (f1 5)))
  (time (dotimes [_ n]
          (f2 5)))
  (time (dotimes [_ n]
          (f3 5)))
  (time (dotimes [_ n]
          (f4 5))))
```

Maybe this leads to a different set of suggestions....

- How about a compromise of `(double -1)`, if we really don't like `.0`?
- I think we want to compose our symbol generator, for anything that is going to represent a double, with

```clj
(fn [sym] (with-meta sym {:tag 'double}))
```

So that we can properly hint to the compiler that the variables are double-valued. I think `(set! *unchecked-math* :warn-on-boxed)` should stay quiet if we do that, and squawk if we don't.

Thoughts?

Reply:
I now feel that it's not the job of this change to tamper with this. If you're happy with everything else, I will reintroduce the double cast of int/long and fix all the unit test expected outputs. For those that differ in output based on the runtime environment, I'll use `#?`.

Reply:
Okay, let's do that - I've made the mistake of using these PR discussions to mix concrete feedback on the issue at hand with wide ranging musing. I agree, let's back this out now and then, armed with some animation etc examples, take another pass to see if there are performance wins we need to handle or if the JVM / browser are in fact smart enough that we get it all for free.